### PR TITLE
Moritz review

### PR DESCRIPTION
**Pull request by Midnighter**
I have tried to improve upon James' suggestions. I've addressed some of his notes in #106. In particular, I've added a paragraph on OPAL.

I think we still need:

- [ ] References for some of the statements in the summary
- [ ] Complete the references that are currently missing
- [ ] Add the figure that James is working on

Let me know what you think about these changes 😃

### CODE REVIEW

**Comment by Midnighter** [comment]
You can get the corresponding PDF from the "Checks" artifacts.

File: paper/paper.md [hunk 1]
```diff
@@ -61,55 +77,89 @@ nf-core community [@ewels_nf-core_2020].
...
 [@bolyen_reproducible_2019] also maintains internally consistent formats for
-storing and processing metagenomic data that new tools can plug into.
+storing and processing metagenomic data that new tools can plug into, however
+this suite of software is aimed mostly at 16S amplicon data [@cite], and thus
+often does have as much support or awareness in 'shotgun' sequencing [@cite]
+metagenomic studies . While some of the taxonomic profilers also come with
+scripts to convert their output into another format and/or merge multiple
+profiles into a single table, such as the Krakentools companion
+package[@lu_metagenome_2022], these are often focused on the specific tool or
+family of tools.
```
Comment (jfy133) [inline paper/paper.md:125]: > thus requiring the user to learn additional tool-specific interfaces.

as in you have to learn the interface of another tool that is only specific to that specific profiler (if that mnakes sense?)

File: paper/paper.md [hunk 2]
```diff
@@ -61,55 +77,89 @@ nf-core community [@ewels_nf-core_2020].
...
+[^7]: <https://www.hdfgroup.org/solutions/hdf5/>
 
 # Acknowledgements
 
-SS was supported by Rapid establishment of comprehensive laboratory pandemic
-preparedness – RAPID-SEQ (Awarded to Prof. Jan Albert). JAFY received funding from the Werner
-Siemens-Stiftung ("Paleobiotechnology", Awarded to Prof. Pierre Stallforth and
+SS was supported by "Rapid establishment of comprehensive laboratory pandemic
+preparedness – RAPID-SEQ" (awarded to Prof. Jan Albert). MB and JAFY were
+supported by the Max Planck Society. JAFY received funding from the Werner
```
Comment (jfy133) [inline paper/paper.md:161]: Need to check if @maxibor is also supported by the WSS Palaeobiotech project and/or the microverse EXC

File: paper/paper.md [hunk 3]
```diff
@@ -61,55 +77,89 @@ nf-core community [@ewels_nf-core_2020].
...
-The QIIME™2 _next-generation microbiome bioinformatics platform_
+metagenomic profiles into that format was entirely left up to the user. The
+format also is not easily loadable into spreadsheet software, and external
+libraries are required for loading the format into data analysis languages such
+as R. The QIIME™2 _next-generation microbiome bioinformatics platform_
 [@bolyen_reproducible_2019] also maintains internally consistent formats for
-storing and processing metagenomic data that new tools can plug into.
+storing and processing metagenomic data that new tools can plug into, however
+this suite of software is aimed mostly at 16S amplicon data [@cite], and thus
+often does have as much support or awareness in 'shotgun' sequencing [@cite]
```
Comment (jfy133) [inline paper/paper.md:120]: For the first citation, I think you can just cite the QIIME1 and QIIME2 papers. They don't mention metagenomics (or rather in the QIIME2 they reference 'inital support for'. For the second, it's hard to cite a non-existance of something... so I think just the two QIIME papers or maybe a link to the docs would be suffient

**Review by jfy133 — COMMENTED** [review]
Some thoughts but I think it reads pretty well now

File: paper/paper.md [hunk 4]
```diff
@@ -61,55 +77,89 @@ nf-core community [@ewels_nf-core_2020].
...
+There exists an initiative to benchmark and compare profilers, as well as
+provide guidance on their fitness for purpose
+[@sczyrba_critical_2017; @meyer_critical_2022]. For that purpose, the
+Open-community Profiling Assessment tooL (OPAL) [@meyer_assessing_2019] was
+developed. Creating a community wide assessment faced many of the challenges
+presented here, however, the chosen solution was to mandate a single output
+format[^8] for all profilers participating in the challenge. Furthermore, OPAL
+is an integrated tool performing assessment and visualization, whereas TAXPASTA
+follows the UNIX philosophy[^9] of doing one thing well, thus allowing for
+composition with other tools.
```
[the remark this answers is not in the document]
Comment (Midnighter) [reply paper/paper.md:106]: > What do you mean by composition with other tools? I thought the idea of that part of the UNIX philosophy was about piping?

That was my thinking, too. Composing analysis workflows by combining different tools. I'm happy to use different wording.

[the remark this answers is not in the document]
Comment (Midnighter) [reply paper/paper.md:106]: > Note I wouldn't have a line break between the OPAL and BIOM formats, as it's the same 'section' of the paper.

You don't think it's a separate paragraph?

[the remark this answers is not in the document]
Comment (jfy133) [reply paper/paper.md:106]: > That was my thinking, too. Composing analysis workflows by combining different tools. I'm happy to use different wording.

It depends what you mean with composition with other tools. We do follow the unix philosphy of do one thing well, but we don't pipe so I would remove the `thus <...> other tools` bit

> You don't think it's a separate paragraph?

Without subheaders (which aren't really wanted in JOSS papers AFAICS), to me that's the section were we compare with other similar tools/formats (biom has it's own CLI IIRC). But this is me being nit picky, fine if you prefer separate.

File: paper/paper.md [hunk 5]
```diff
@@ -61,55 +77,89 @@ nf-core community [@ewels_nf-core_2020].
...
 # Acknowledgements
 
-SS was supported by Rapid establishment of comprehensive laboratory pandemic
-preparedness – RAPID-SEQ (Awarded to Prof. Jan Albert). JAFY received funding from the Werner
-Siemens-Stiftung ("Paleobiotechnology", Awarded to Prof. Pierre Stallforth and
+SS was supported by "Rapid establishment of comprehensive laboratory pandemic
+preparedness – RAPID-SEQ" (awarded to Prof. Jan Albert). MB and JAFY were
+supported by the Max Planck Society. JAFY received funding from the Werner
+Siemens-Stiftung ("Paleobiotechnology", awarded to Prof. Pierre Stallforth and
 Prof. Christina Warinner).
```
Comment (maxibor) [inline paper/paper.md:163]:
```suggestion
Prof. Christina Warinner), and MB received funding from the Balance of Microverse
Cluster of Excellence (awared to Prof. Christina Warinner).
```

File: paper/paper.md [hunk 6]
```diff
@@ -15,42 +15,58 @@ authors:
...
-performing)? The nature of this field is such that it intersects with ecology,
-statistics, and bioinformatics.
-
-In part, due to the interdisciplinary nature of the field, there exists a
-diverse number of bioinformatics tools in order to analyse metagenomic
-sequencing data and produce metagenomic profiles; answering the question of
-who is there and estimate the relative abundance. Most of these tools have developped
-their own tabular result format, which complicates downstream analysis and in
-particular comparison across tools remains challenging.
+Metagenomic analysis is chiefly concerned with untargeted genetic
```
[the remark this answers is not in the document]
Comment (maxibor) [reply paper/paper.md:43]: As a non native english reader, I had to google this. Maybe rephrase to something like this ?
```suggestion
One of the main goal of metagenomics is the untargeted genetic
```

File: paper/paper.md (hunk 2 above)
Comment (maxibor) [reply paper/paper.md:161]: I made a suggestion below

File: paper/paper.md (hunk 5 above)
Comment (jfy133) [reply paper/paper.md:163]: Isn't tehre a full version with some ID...?

from AMDirT: 

> M.B. and A.H. were supported under Germany’s Excellence Strategy EXC 2051 (Project-ID 390713860, “Bal-
ance of the Microverse”).

Comment (maxibor) [reply paper/paper.md:163]: I updated it, but looks like it isn't always necessary

File: paper/paper.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Midnighter) [reply paper/paper.md:106]: @jfy133 do you think that we **should** allow for streaming output such that piping becomes possible? It would be fairly straight forward to allow for this.

[the remark this answers is not in the document]
Comment (jfy133) [reply paper/paper.md:106]: I guess you could for stdout piping although I don't really see the utility much unless the output is in long format (wide you would lose the header from the output from merge)... And even then I can't really think of when you would use it in an analysis workflow...

I can't see how you would implement it for stdin though as you wouldn't have the sample ID info that in most cases is stored in the file name

But if it would take you 20 minutes I guess why not (??) But any longer would not be worth it

[the remark this answers is not in the document]
Comment (Midnighter) [reply paper/paper.md:106]: I have adjusted the sentence for now, but it's definitely something we can consider.

**Comment by Midnighter** [comment]
Thank you all for your comments. I'm quite happy with this version for now, so I will merge it and comment on the review. If we need to tweak it further, I will put it on HackMD.
